crates/fluvio-protocol/src/record/batch.rs

```diff
@@ -917,4 +928,22 @@ mod test {
         batch.header.set_schema_id();
         assert!(batch.header.has_schema());
     }
+
+    #[test]
+    fn test_truncate_uncompleated_raw_records_in_memory_records() {
+        // then
+        let mem_records = vec![Record::default(), Record::default(), Record::default()];
+        let batch = Batch::from(mem_records.clone());
+
+        // when
+        let mut batch_raw_records: Batch<RawRecords> = Batch::try_from(batch).unwrap();
+        let mut other = bytes::BytesMut::from(batch_raw_records.records.0.clone());
+        other.put_slice(&[0, 1, 2, 3]); // add random bytes to be handle as uncompleated
+        batch_raw_records.records.0 = other.freeze();
+        let batch_result = batch_raw_records.memory_records();
+
+        // then
+        let batch = batch_result.unwrap();
+        assert_eq!(batch.len(), 3);
```

Comment:
check offset and record value.
also add test case for compressed

Reply:
sure, I added record value check. 

> also since this covers incomplete batches, would need cover that scenario as well.
Actually, this test is doing it `other.put_slice(&[0, 1, 2, 3]);` is adding an incomplete record at the end